Tests/govuk_ios/govuk_ios_unit_tests/Specs/Coordinators/AnalyticsConsentCoordinatorTests.swift

```diff
@@ -0,0 +1,37 @@
+import XCTest
+
+@testable import govuk_ios
+
+final class AnalyticsConsentCoordinatorTests: XCTestCase {
+    func test_start_hasSeenAnalyticsConsent_callsDismiss() throws {
+        let analyticsConsentService = MockAnalyticsConsentService()
+        analyticsConsentService._stubbedHasSeenAnalyticsConsent = true
+        let mockNavigationController = UINavigationController()
+        let expectation = expectation()
+        let sut = AnalyticsConsentCoordinator(
+            navigationController: mockNavigationController,
+            analyticsConsentService: analyticsConsentService,
+            analyticsService: MockAnalyticsService(),
+            dismissAction: {
+                expectation.fulfill()
+            }
+        )
+        sut.start()
+        wait(for: [expectation], timeout: 1)
+    }
+
+    func test_start_hasNotSeenAnalyticsConsent_doesNotCallDismiss() throws {
+        let analyticsConsentService = MockAnalyticsConsentService()
+        analyticsConsentService._stubbedHasSeenAnalyticsConsent = false
+        let mockNavigationController = UINavigationController()
+        let sut = AnalyticsConsentCoordinator(
+            navigationController: mockNavigationController,
+            analyticsConsentService: analyticsConsentService,
+            analyticsService: MockAnalyticsService(),
+            dismissAction: {
+                XCTFail()
```

Comment:
You could do an inverse expectation here to make sure that the `XCTFail` isn't being called after the test has completed

Reply:
Done